src/test/change_stream.rs

```diff
@@ -709,3 +710,68 @@ async fn create_coll_pre_post() -> Result<()> {
 
     Ok(())
 }
+
+// Prose test 19: large event splitting
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn split_large_event() -> Result<()> {
+    let _guard = LOCK.run_concurrently().await;
+
+    let client = Client::test_builder().build().await;
+    if !client.server_version_gte(7, 0) {
+        log_uncaptured(format!(
+            "skipping change stream test on unsupported version {:?}",
+            client.server_version
+        ));
+        return Ok(());
+    }
+    if !client.is_replica_set() && !client.is_sharded() {
+        log_uncaptured("skipping change stream test on unsupported topology");
+        return Ok(());
+    }
+
+    let db = client.database("change_stream_tests");
+    db.collection::<Document>("split_large_event")
+        .drop(None)
+        .await?;
+    db.create_collection(
+        "split_large_event",
+        CreateCollectionOptions::builder()
+            .change_stream_pre_and_post_images(ChangeStreamPreAndPostImages { enabled: true })
+            .build(),
+    )
+    .await?;
+
+    let coll = db.collection::<Document>("split_large_event");
+    coll.insert_one(doc! { "value": "q".repeat(10 * 1024 * 1024) }, None)
+        .await?
+        .inserted_id;
+    let stream = coll
+        .watch(
+            [doc! { "$changeStreamSplitLargeEvent": {} }],
+            ChangeStreamOptions::builder()
+                .full_document_before_change(Some(FullDocumentBeforeChangeType::Required))
+                .build(),
+        )
+        .await?
+        .with_type::<Document>();
```

Comment:
Unfortunately, the stream needs to produce `Document`s rather than `ChangeStreamEvent`s because the fields are arbitrarily split up across the parts, so there are no guarantees that required fields are present in any particular fragment.  This also means that there's no need to add the `splitEvent: { fragment: 1, of: 2 }` structure to `ChangeStreamEvent` (and it seems like doing so would be misleading, since you couldn't actually use that type for these events).